evm/precompile/namespace/namespace.go

```diff
@@ -0,0 +1,53 @@
+package namespace
+
+import (
+	"context"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethprecompile "pkg.berachain.dev/polaris/eth/core/precompile"
+	"pkg.world.dev/world-engine/evm/precompile/contracts/bindings/cosmos/precompile/namespace"
+	"pkg.world.dev/world-engine/evm/x/namespace/types"
+)
+
+const name = "world_engine_namespace_registrar"
+
+type Contract struct {
+	ethprecompile.BaseContract
+	types.MsgServer
+	types.QueryServiceServer
+}
+
+func NewPrecompileContract(ms types.MsgServer, qs types.QueryServiceServer) *Contract {
+	return &Contract{
+		BaseContract: ethprecompile.NewBaseContract(
+			namespace.NamespaceMetaData.ABI,
+			common.BytesToAddress(authtypes.NewModuleAddress(name)),
+		),
+		MsgServer:          ms,
+		QueryServiceServer: qs,
+	}
+}
+
+func (c *Contract) Register(
+	ctx context.Context,
+	namespace string,
+	grpcAddress string,
+) (bool, error) {
+	_, err := c.UpdateNamespace(ctx, &types.UpdateNamespaceRequest{
+		Authority: "",
+		Namespace: &types.Namespace{
+			ShardName:    namespace,
+			ShardAddress: grpcAddress,
+		},
+	})
+	// if err == nil, this returns true, which means it succeeded.
+	return err == nil, err
```

Comment:
It would be more natural to just return err here and not bother with the bool. Is the (bool, error) required because of some external interface?

Reply:
yeah, polaris precompile requires another return value other than error